### ISSUE TITLE
False Positive with result_unit_err when using type aliases

### ISSUE THREAD

**Issue by pchampin** [rust-lang/rust-clippy#6546]
Lint name: result_unit_err

I tried this code:

```rust
type MyRes<A, B> = Result<(A, B), Box<dyn std::error::Error>>;

pub fn foo(x: i32) -> MyRes<i32, ()> {
    Ok((x, ()))
}
```

I expected to see this happen: no warning

Instead, this happened: clippy issues a warning about `result_unit_err`. It seems to naively assume that the 2nd parameter of a type alias of `Result` is always the error type.

### Meta

- `cargo clippy -V`: `clippy 0.0.212 (e1884a8 2020-12-29)`
- `rustc -Vv`:
  ```
  rustc 1.49.0 (e1884a8e3 2020-12-29)
  binary: rustc
  commit-hash: e1884a8e3c3e813aada8254edfa120e85bf5ffca
  commit-date: 2020-12-29
  host: x86_64-unknown-linux-gnu
  release: 1.49.0
  ```
